Feat: Решетка на балках

Kirozet opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!-- Include a small to medium description of what your PR changes. Document every changes or this may delay review or even discourage maintainers from merging your PR! -->
Позволяет возможность ставить сетку из прутьев на балки в космосе.
## Why It's Good For The Game
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Images of changes
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif of your feature if you want -->

## Changelog
:cl:
add: Добавлена возможность собирать решетку из прутьев не только на полу, но и в космосе, но только на балках и дочерних структурах.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Kirozet committed:
Update rods.dm

Добавлена переменная on_latice в крафт

Kirozet committed:
Добавлена проверка на наличие балок

Добавлена проверка на наличие балок под игроком, если on_lattice = 1

Kirozet committed:
Введена переменная on_latice

Переменная on_latice нужна для разрешения установки структуры на балках, если запрещена установка в космосе.

Kirozet commented:
в коммитах latice с одной буквой t, но в коде всё нормально. Я просто тапочек и уже утомился заполняя это коммиты

Vallat commented on code/game/objects/items/stacks/stack.dm:
```diff
@@ -196,8 +196,12 @@
 		if(R.on_floor && !istype(usr.drop_location(), /turf/simulated))
-			to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
-			return FALSE
+			if(R.on_lattice && (!locate(/obj/structure/lattice) in usr.drop_location() ))
+				to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor or lattice!</span>")
+				return FALSE
+			if(!R.on_lattice)
+				to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
+				return FALSE
```
```suggestion
			if(!R.on_lattice)
				to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
				return FALSE
			if(!(locate(/obj/structure/lattice) in usr.drop_location()))
				to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor or lattice!</span>")
				return FALSE
```

Kirozet replied:
?? Будет выводиться `to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor or lattice!</span>")`  даже если в крафте не стоит `on_lattice = 1` и постройка на балках не подразумевается

Vallat replied:
Не будет, в этом случае оно завершится после первого условия `if(!R.on_lattice)` и выведет `to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")`

Kirozet replied:
понял, но... смысл если результат один?

Vallat replied:
Избавление от ненужных проверок, более чистый и логически понятный код. Мы теперь сначала проверяем находимся ли мы на полу, если нет, проверяем можем ли мы строиться на прутьях, если можем, то проверяем уже наличие этих прутьев.

Kirozet committed:
Update code/game/objects/items/stacks/stack.dm

Избавление от ненужных проверок, более чистый и логически понятный код. Мы теперь сначала проверяем находимся ли мы на полу, если нет, проверяем можем ли мы строиться на прутьях, если можем, то проверяем уже наличие этих прутьев.

Co-authored-by: Vallat <vox.crit2013@gmail.com>
